lib/pbench/server/indexing_tarballs.py

```diff
@@ -234,15 +241,38 @@ def sigquit_handler(*args):
 
                         ptb = None
                         try:
+                            path = os.path.realpath(tb)
+
                             # "Open" the tar ball represented by the tar ball object
                             idxctx.logger.debug("open tar ball")
                             ptb = PbenchTarBall(
                                 idxctx,
-                                os.path.realpath(tb),
+                                path,
                                 tmpdir,
                                 Path(self.incoming, controller),
                             )
 
+                            try:
+                                dataset = Dataset.attach(
+                                    path=path,
+                                    state=States.INDEXING,
+                                )
+                            except DatasetNotFound:
+                                idxctx.logger.warn(
+                                    "Unable to locate Dataset {}", path,
```

Comment:
Can we make these two conditions errors that don't proceed with indexing?  Seems like we don't want to proliferate bad state.

Reply:
My first pass was to splice the DB tracker on top of the existing without breaking the flows that currently depend entirely on file system state. At some point, yeah, we definitely want consistency checks here. My thought was to defer that for now, then build consistency checks into the audit tool, and think about what we want to fail and what point if there are inconsistencies.

I can add that in the current PR, but that means new unit testing scenarios and such; I'd rather separate that, but I'll create an issue.

Reply:
Perhaps a `FIXME` in the code would suffice?  I support the idea of avoiding mission creep, but I suspect that capturing the details of this request without annotating the code itself will be quite a challenge.